readme.md

```diff
@@ -1,12 +1,105 @@
+
+The next major release of the driver (v5) will drop support for callbacks.
+This package will also have a major release at that time to update the dependency requirement to `^5.0.0`.
+Users can expect to be able to upgrade to `v5` adopting the changes and features shipped in that version while using this module to maintain any callback code they still need to work on migrating.
+
+## [API](https://mongodb.github.io/node-mongodb-native/)
+
+The API is inherited from the driver, which is [documented here](https://mongodb.github.io/node-mongodb-native/).
+If it is relevant to inspect the precise differences, you can peruse the [type definitions of wrapped APIs](https://github.com/mongodb-js/nodejs-mongodb-legacy/blob/main/mongodb-legacy.d.ts).
+
+The wrappers are implemented as subclasses of each of the existing driver's classes with extra logic to handle the optional callback behavior. And all other driver exports are re-exported directly from the wrapper. This means any new APIs added to the driver will be automatically pulled in as long as the updated driver is installed.
+
+Take this hypothetical example:
+
+```ts
+// Just add '-legacy' to my mongodb import
+import { MongoClient } from 'mongodb-legacy';
+const client = new MongoClient();
+const db = client.db();
+
+const collection = db.collection('pets');
+// Returns an instance of LegacyFindCursor which is a subclass of FindCursor
+const dogCursor = collection.find({ kind: 'dog' });
+// Using .next with callbacks still works!
+dogCursor.next((error, dog) => {
+  if (error) return handling(error);
+  console.log(dog);
+});
+// Brand new api that pets all dogs! (does not support callbacks)
+dogCursor.petAll().then(result => {
+  console.log('all dogs got pats!');
+});
 ```
 
-We are soon shipping our `'mongodb'` with promise only APIs, but this package can help those who have difficulty adopting that change by continuing to offer our API in it's existing form a combination of callback and promise support. We hope that this module will require only the change of the import string from `'mongodb'` to `'mongodb-legacy'` along with adding `'mongodb-legacy'` to your `package.json`. Our intent is to ensure that the existing APIs offer precisely the same behavior as before, the logic for handling callback or promise been moved to these light wrappers. Please let us know if you encounter any differences if you have need of this package.
+> NOTE: The `petAll()` api is brand new in this example and will not support an optional callback. If adopting this API is deep inside code that already relies on eventually calling a callback to indicate the end of an operation it is possible to use `.then`/`.catch` chains to handle the cases that are needed. We recommend offloading this complexity to node's [callbackify utility](https://nodejs.org/dist/latest-v16.x/docs/api/util.html#utilcallbackifyoriginal): `callbackify(() => dogCursor.petAll())(callback)`
```

Comment:
Just double checking that this still works for class methods (without binding `this`)

Reply:
Yep, one could use bind, but it's a bit ugly and creates a new bound function, using an arrow function accomplished the same thing you're calling the function as you normally do and passing the promise result to callbackify